RELEASE.md

```diff
@@ -0,0 +1,38 @@
+# Lodestar Release Guidelines
+
+![lodestar-release](docs/images/lodestar-releases.png)
+
+## Release Process Rules
+
+1. The team selects a commit from `master` as a "stable target" for a new version release.
+2. The selected commit is branched into a separate release branch (Example: `v0.33.x`) 
+3. The commit on the separate release branch is tagged with `beta.x`. (Example: `v0.33.0-beta.0`) and published for testing as a pre-release.
+4. The team bumps `master` to the next version and continues releasing nightly builds. (Example: `v0.34.0`)
```

Comment:
Suggestion: maybe make it explicit this is about bumping the version as specified in `package.json`. And nothing to do with branching or tagging etc.

Reply:
Made it more explicit in the new commit.